File: lingting-core/src/main/java/live/lingting/framework/function/ThrowingFunction.java
package live.lingting.framework.function;

/**
 * @author lingting 2023/2/2 17:36
 */
@FunctionalInterface
@SuppressWarnings("java:S112")
public interface ThrowingFunction<T, R> extends ThrowableFunction<T, R> {

	R apply(T t) throws Exception;

}

File: lingting-core/src/main/kotlin/live/lingting/framework/function/ThrowingFunction.kt
package live.lingting.framework.function

/**
 * @author lingting 2023/2/2 17:36
 */
fun interface ThrowingFunction<T, R> : ThrowableFunction<T, R> {

    @Throws(Exception::class)
    override fun apply(t: T): R

}
